protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,306 @@
+
+If reducing the `commitment amount` then only funds contained within the AMMs `general` account are eligible for removal. If the deduction is less than the `general` account's balance then the reduced funds will be removed immediately and the AMM will enter `single-sided` mode as specified above to reduce the position. If a deduction of greater than the `general` account is requested then the transaction is rejected and no changes are made.
+
+
+#### Cancellation
+
+In addition to amending to reduce the size a user may also cancel their AMM entirely. In order to do this they must submit a transaction containing only a field `Reduction Strategy` which can take two values:
+
+ - `Abandon Position`: In this case, any existing position the AMM holds is given up to the network to close as a liquidation. This is performed in two steps:
+   - All funds in the AMM's `general` account are transferred back to the party who created the AMM.
+   - The position is marked as requiring liquidation and is taken over by the network through the usual liquidation processes. All funds in the margin account are transferred to the network's insurance pool as in a forced liquidation
+ - `Reduce-Only`: This moves the AMM to a reduce-only state, in which case the position is reduced over time and ranges dynamically update to ensure no further position is taken. As such:
+   - If the AMM is currently short, the `lower bound` is removed
+   - If the AMM is currently long, the `upper bound` is removed
+   - The `upper`/`lower` bound (if the AMM is currently short/long) is then set to the AMM's current `fair price`. In this mode the AMM should only ever quote on the side which will reduce it's position (it's `upper`/`lower` bound should always be equal to the current `fair price` belief).
+   - Once the position reaches `0` the AMM can be cancelled and all funds in the general account can be returned to the creating party
+   - This acts similarly to the mode when an AMM is synchronising, except that the position will be closed in pieces as the price moves towards the `base price` rather than all at once at the nearest price.
+
+Note that, whilst an `Abandon Position` transaction immediately closes the AMM a `Reduce-Only` transaction will keep it active for an uncertain amount of time into the future. During this time, any Amendment transactions received should move the AMM out of `Reduce-Only` mode and back into standard continuous operation.
+
```

Comment:
The implementation allows going from `Abandon Position` to `Reduce-Only`, which would be nice to reference in the spec.

Reply:
I'm not sure how that would be possible given `Abandon Position` should totally close the position?

Reply:
Sorry @TomMcL, I meant the other way around, from reduce to abandon.